access violation in wxFileName() with invalid path

If wxFileName constructor is feeded with invalid path, say
"file:/abc/def.txt".
It will access invalid memory location when it try to 
read path[0] because path is empty.

diff --git a/wxWidgets/src/common/filename.cpp b/wxWidgets/src/common/filename.cpp
--- a/wxWidgets/src/common/filename.cpp
+++ b/wxWidgets/src/common/filename.cpp
@@ -368,16 +368,21 @@
     }
 
     format = GetFormat( format );
 
     // 0) deal with possible volume part first
     wxString volume,
              path;
     SplitVolume(pathOrig, &volume, &path, format);
+    if ( path.empty())
+    {
+        // not a valid path
+        return;
+    }
     if ( !volume.empty() )
     {
         m_relative = false;
 
         SetVolume(volume);
     }
 
     // 1) Determine if the path is relative or absolute.
